### ISSUE TITLE
Special case binding to class property.

### ISSUE THREAD

**Issue by mhevery** [angular/angular#505]
Binding to `class` attribute should result to binding to `classList` and `classNames` properties

```
<div class="foo" [class]="expr1" [class.bar]="expr2">
```
should result of merging of `foo` and `expr1` and `bar` class names.

Considerations:
1. `class` properties should be considered fixed. So when `[class]` changes it should merge into existing `class` names rather than override them.
2. `epr1` could have several formats:
  -  space delimited strings: `"baz bar"`
  - An array with names `["baz", "bar"]`
  - A hash map with keys becoming class names only if the value is truthy in the given platform. `{"baz": true, "bar": false}`
3. `expr2` evaluates to truthy in the given platform. It then sets or removes the `bar` class name.

If there is a collision, than they are resolved in the same order. 1) `class` 2) `[class]` 3) `[class.name]`

**Comment by naomiblack** [comment]
Need to also do this for style

**Comment by pkozlowski-opensource** [comment]
* `[class.classname]="expr"` support was added in 7ce4f66cdcd030132d16c46b66ffd0f0ea5b008b
* `[style.stylename]="expr"` and `[style.stylename.suffix]="expr"` support was added in ee3f709fbf4d578a34eee7255d381d4f6c1dccca

What remains is support for `[class]="expr"` and `[style]="expr"`

**Comment by mhevery** [comment]
Moving to M5 because we are blocked on Structural Changes

**Comment by vsavkin** [comment]
The support for Pipes have been added. This issue should be unblocked now.

**Comment by pkozlowski-opensource** [comment]
@vsavkin yup, saw your commits, working on the topic right now! Could you please review #771 as I'm going to need a pipe for objects (or at least I'm persuaded I'm going to need those!)

**Comment by mhevery** [comment]
Closing since https://github.com/angular/angular/issues/876 covers the missing use case and the rest is already in.
